cub/cub/thread/thread_load.cuh

```diff
@@ -190,48 +190,44 @@ _CCCL_DEVICE _CCCL_FORCEINLINE void UnrolledCopy(RandomAccessIterator src, T* ds
-      return retval;                                                                                       \
-    }                                                                                                      \
-    template <>                                                                                            \
-    _CCCL_DEVICE _CCCL_FORCEINLINE uint2 ThreadLoad<cub_modifier, uint2 const*>(uint2 const* ptr)          \
-    {                                                                                                      \
-      uint2 retval;                                                                                        \
-      asm volatile("ld." #ptx_modifier ".v2.u32 {%0, %1}, [%2];"                                           \
-                   : "=r"(retval.x), "=r"(retval.y)                                                        \
-                   : _CUB_ASM_PTR_(ptr));                                                                  \
-      return retval;                                                                                       \
-    }                                                                                                      \
-    template <>                                                                                            \
-    _CCCL_DEVICE _CCCL_FORCEINLINE unsigned long long ThreadLoad<cub_modifier, unsigned long long const*>( \
-      unsigned long long const* ptr)                                                                       \
-    {                                                                                                      \
-      unsigned long long retval;                                                                           \
-      asm volatile("ld." #ptx_modifier ".u64 %0, [%1];" : "=l"(retval) : _CUB_ASM_PTR_(ptr));              \
-      return retval;                                                                                       \
+#  define _CUB_LOAD_8(cub_modifier, ptx_modifier)                                                              \
+    template <>                                                                                                \
+    _CCCL_DEVICE _CCCL_FORCEINLINE ushort4 ThreadLoad<cub_modifier, ushort4 const*>(ushort4 const* ptr)        \
+    {                                                                                                          \
+      ushort4 retval;                                                                                          \
+      asm volatile("ld." #ptx_modifier ".v4.u16 {%0, %1, %2, %3}, [%4];"                                       \
+                   : "=h"(retval.x), "=h"(retval.y), "=h"(retval.z), "=h"(retval.w)                            \
+                   : "l"(ptr));                                                                                \
```

Comment:
No change requested, but this is definitely something we would want to be handled in <cuda/ptx>

@ahendriksen what do you think how much of an effort would that be?